addons/point_of_sale/static/src/js/models.js

```diff
@@ -3201,6 +3202,27 @@ class Order extends PosModel {
             return true;
         }
     }
+    _get_qr_code_data() {
+        if (this.pos.company.point_of_sale_use_ticket_qr_code) {
+            const codeWriter = new window.ZXing.BrowserQRCodeSvgWriter();
+            let port = ''
+            if (window.location.port) {
+                port = `:${window.location.port}`
+            }
+            const address = `${window.location.protocol}//${window.location.hostname}${port}/pos/ticket/validate?order_reference=${this.uid}&ticket_amount=${this.get_total_with_tax()}`
+            let qr_code_svg = new XMLSerializer().serializeToString(codeWriter.write(address, 150, 150));
+            return "data:image/svg+xml;base64,"+ window.btoa(qr_code_svg);
+        } else {
+            return false;
+        }
+    }
+    _get_qr_ticket_address() {
```

Comment:
You should not do this kind of construction, you should use 'web.base_url' stored in system parameters

Reply:
As it's in the fronted, I don't see a way to get it (unless we add a controller for it).
But I see window.location.origin being used at various places to get the origin (also, in this same file) so would this be ok with you?

Reply:
we do not need a controller each time we want to access data from frontend. When you are connected as a user, you can call functions on models you have access to. In POS, all data needed to run it are loaded at the beginning in 'load_pos_data' function in python which is called in models.js